src/_gettsim/lohnsteuer/lohnsteuer.py

```diff
@@ -2,68 +2,240 @@
+        thresholds=thresholds, intercepts=intercepts, rates=rates
+    )
+
+    return parameter_max_lohnsteuer_klasse_5_6
+
+
+@policy_function(start_date="2015-01-01")
+def basistarif(
+    einkommen_y: float,
+    einkommensteuer__parameter_einkommensteuertarif: PiecewisePolynomialParamValue,
+) -> float:
+    """Lohnsteuer in the Basistarif."""
+    return piecewise_polynomial(
+        einkommen_y, einkommensteuer__parameter_einkommensteuertarif
+    )
+
+
+@policy_function(start_date="2015-01-01")
+def splittingtarif(
     einkommen_y: float,
     einkommensteuer__parameter_einkommensteuertarif: PiecewisePolynomialParamValue,
+) -> float:
+    """Lohnsteuer in the Splittingtarif."""
+    return 2 * piecewise_polynomial(
+        einkommen_y / 2, einkommensteuer__parameter_einkommensteuertarif
+    )
+
+
+@policy_function(start_date="2015-01-01")
+def tarif_klassen_5_und_6(
+    einkommen_y: float,
+    einkommensteuer__parameter_einkommensteuertarif: PiecewisePolynomialParamValue,
+    parameter_max_lohnsteuer_klasse_5_6: PiecewisePolynomialParamValue,
+) -> float:
+    """Lohnsteuer for Lohnsteuerklassen 5 and 6."""
+
+    lohnsteuer_klasse_5_6 = basis_für_klassen_5_6(
+        einkommen_y, einkommensteuer__parameter_einkommensteuertarif
+    )
+    max_lohnsteuer = piecewise_polynomial(
+        einkommen_y, parameter_max_lohnsteuer_klasse_5_6
+    )
+    min_lohnsteuer = (
+        einkommensteuer__parameter_einkommensteuertarif.rates[0, 1] * einkommen_y
+    )
+    return np.minimum(np.maximum(min_lohnsteuer, lohnsteuer_klasse_5_6), max_lohnsteuer)
+
+
+@policy_function(start_date="2015-01-01")
+def betrag_m(
     steuerklasse: int,
-    einkommensgrenzwerte_steuerklassen_5_6: dict[int, float],
+    basistarif: float,
+    splittingtarif: float,
+    tarif_klassen_5_und_6: float,
 ) -> float:
     """Withholding tax on earnings (Lohnsteuer)"""
-    return lohnsteuerformel(
-        einkommen_y=einkommen_y,
-        parameter_einkommensteuertarif=einkommensteuer__parameter_einkommensteuertarif,
-        einkommensgrenzwerte_steuerklassen_5_6=einkommensgrenzwerte_steuerklassen_5_6,
-        steuerklasse=steuerklasse,
+
+    if steuerklasse == 1 or steuerklasse == 2 or steuerklasse == 4:
+        out = basistarif
+    elif steuerklasse == 3:
+        out = splittingtarif
+    else:
+        out = tarif_klassen_5_und_6
+    out = out / 12
+    return max(out, 0.0)
+
+
+@policy_function(start_date="2015-01-01")
+def basistarif_mit_kinderfreibetrag(
+    einkommen_y: float,
+    einkommensteuer__parameter_einkommensteuertarif: PiecewisePolynomialParamValue,
+    kinderfreibetrag_soli_y: float,
+) -> float:
+    """Lohnsteuer in the Basistarif deducting the Kindefreibetrag."""
+    einkommen_abzüglich_kinderfreibetrag_soli = np.maximum(
+        einkommen_y - kinderfreibetrag_soli_y, 0
+    )
+    return piecewise_polynomial(
+        einkommen_abzüglich_kinderfreibetrag_soli,
```

Comment:
```suggestion
        x=einkommen_abzüglich_kinderfreibetrag_soli,
```
Please call by name wherever n arguments > 1. Apologies for being pedantic here.